samigo/samigo-app/src/java/org/sakaiproject/jsf/util/SamigoJsfTool.java

```diff
@@ -360,7 +363,16 @@ else if (target.indexOf("/jsf/evaluation/totalScores") > -1
 			bean.setAttachment((Long) toolSession.getAttribute("assessmentGradingId"));
 			toolSession.removeAttribute("SENT_TO_FILEPICKER_HELPER");
 		}
-    
+      else if (target.indexOf("/jsf/evaluation/gradeStudentResult") > -1
+				&& ContextUtil.lookupParam("resetCache") != null && ContextUtil.lookupParam("resetCache").equals("true")) {
```

Comment:
To only call `lookupParam` once:
```suggestion
				&& Boolean.TRUE.toString().equals(ContextUtil.lookupParam("resetCache"))) {
```